openssl/src/ec.rs

```diff
@@ -459,12 +459,35 @@ impl EcKeyBuilderRef {
     pub fn generate_key(&mut self) -> Result<&mut EcKeyBuilderRef, ErrorStack> {
         unsafe { cvt(ffi::EC_KEY_generate_key(self.as_ptr())).map(|_| self) }
     }
+
+    /// Sets the public key based on affine coordinates.
+    pub fn set_public_key_affine_coordinates(&mut self,
+                                             x: &mut BigNumRef,
```

Comment:
My reading of how the implementation calls out within OpenSSL (`EC_POINT_set_affine_coordinates_GFp` /`EC_POINT_set_affine_coordinates_GF2m`) is that no modification will be made to `x` / `y` and my local checks backed that up.

I changed the Rust signature out of an abundance of caution. However, if the approach has previously been to make the Rust level more reflective of what is actually going on in OpenSSL regardless of the C API I am happy with that.

Let me know your preference here and I'll fix this up.